drivers/flash/soc_flash_stm32f4.c

```diff
@@ -255,6 +254,12 @@ static struct flash_driver_api flash_stm32f4_api = {
 static int stm32f4_flash_init(struct device *dev)
 {
 	struct flash_priv *p = dev->driver_data;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(sector); i++) {
+		sector[i].start = CONFIG_FLASH_BASE_ADDRESS +
+			i * sector[i - 1].len;
+	}
```

Comment:
This is confusing, since the start address of the sector can change depending on the base address used. Shouldn't this driver simply hardcode the list of sectors with the respective base address?

We need the app to be able to write and read into any place of the flash, doesn't matter where the zephyr app is located.